lib/upgrade/converters/policy_test.go

```diff
@@ -302,12 +303,70 @@ var policyTable = []struct {
 				Egress:         []apiv3.Rule{V3EgressRule1},
 				Selector:       "thing == 'value'",
 				DoNotTrack:     true,
-				PreDNAT:        true,
+				PreDNAT:        false,
 				ApplyOnForward: true,
 				Types:          []apiv3.PolicyType{apiv3.PolicyTypeIngress, apiv3.PolicyTypeEgress},
 			},
 		},
 	},
+	{
+		description: "policy with preDNAT policy and types contains egress is converted in v3 API",
+		v1API: &apiv1.Policy{
+			Metadata: apiv1.PolicyMetadata{
+				Name: "dontcare",
+			},
+			Spec: apiv1.PolicySpec{
+				Order: &order1,
+				IngressRules: []apiv1.Rule{apiv1.Rule{
+					Action: "allow",
+					Source: apiv1.EntityRule{
+						Nets: []*net.IPNet{&cidr1, &cidr2},
+					},
+					Destination: apiv1.EntityRule{},
+				}},
+				Selector: "role in { 'value1', 'cake' }",
+				PreDNAT:  true,
+				Types:    []apiv1.PolicyType{apiv1.PolicyTypeIngress},
+			},
+		},
+		v1KVP: &model.KVPair{
+			Key: model.PolicyKey{
+				Name: "dontcare",
+			},
+			Value: &model.Policy{
+				Order: &order1,
+				InboundRules: []model.Rule{model.Rule{
+					Action:  "allow",
+					SrcNets: []*net.IPNet{&cidr1Net, &cidr2Net},
+				}},
+				OutboundRules:  []model.Rule{},
+				Selector:       "role in { 'value1', 'cake' }",
+				PreDNAT:        true,
+				ApplyOnForward: true,
+				Types:          []string{"ingress"},
```

Comment:
Unfortunately I don't think this is actually testing your fix now.  You need a test case with 'ingress' and 'egress' here (in v1KVP), but with only 'ingress' after conversion to v3 API.